### PR TITLE
Misc fixes

### PR DESCRIPTION
**Pull request by sougou**
This has a bunch of misc fixes. Each CL has its own fix.

### CODE REVIEW

File: go/vt/dbconnpool/connection.go [hunk 1]
```diff
@@ -36,24 +37,21 @@ func (dbc *DBConnection) handleError(err error) {
 
 // ExecuteFetch is part of PoolConnection interface.
 func (dbc *DBConnection) ExecuteFetch(query string, maxrows int, wantfields bool) (*proto.QueryResult, error) {
-	start := time.Now()
+	defer dbc.mysqlStats.Record("Exec", time.Now())
```
Comment (alainjobart) [inline go/vt/dbconnpool/connection.go]: Does that work? I thought the eval of time.Now() would be done at end of function time, not right now...

Comment (enisoc) [inline go/vt/dbconnpool/connection.go]: """
Each time a "defer" statement executes, the function value and parameters to the call are evaluated as usual and saved anew but the actual function is not invoked.
"""
https://golang.org/ref/spec#Defer_statements

Comment (alainjobart) [inline go/vt/dbconnpool/connection.go]: Great then. i guess I was confused with:
defer f() {
  XXX
}
where XXX is evaluated when the function runs at the end, not at declaration time.

**Comment by alainjobart** [comment]
LGTM

**Pull request merged**